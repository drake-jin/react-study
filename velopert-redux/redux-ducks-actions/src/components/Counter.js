import React from 'react';
import PropTypes from 'prop-types';
import './Counter.css';

const Counter = ({number, color, index, onIncrement, onDecrement, onSetColor}) => {
    console.log(color)
    return (
        <div 
            className="Counter" 
            onClick={()=>onIncrement(index)} 
            onContextMenu={
                (e) => { 
                    e.preventDefault(); 
                    onDecrement(index);
                }
            } 
            onDoubleClick={()=>onSetColor(index)}
            style={{backgroundColor: color}}>
                {number}
        </div>
    );
};

Counter.propTypes = {
    number: PropTypes.number,
    index: PropTypes.number,
    color: PropTypes.string,
    onIncrement: PropTypes.func,
    onDecrement: PropTypes.func,
    onSetColor: PropTypes.func
};

Counter.defaultProps = {
    number: 0,
    color: 'black',
    index: 0,
    onIncrement: () => console.warn('onIncrement not defined'),
    onDecrement: () => console.warn('onDecrement not defined'),
    onSetColor: () => console.warn('onSetColor not defined')
};

export default Counter;