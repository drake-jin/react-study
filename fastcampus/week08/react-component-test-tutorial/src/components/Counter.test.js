import React from 'react'
import renderer from 'react-test-renderer'
import Counter from './Counter'

describe('src/components/Counter.js', () => {
  let component = null;

  it('renders correctly', () => {
    component = renderer.create(<Counter />)
  })

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  it('increase correctly', () => {
    component.getInstance().onIncrease()
    expect(component.getInstance().state.value).toBe(2);
    // count increase한 값이 2인지 확인
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('decrease correctly', () =>{
    component.getInstance().onDecrease()
    expect(component.getInstance().state.value).toBe(1)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})