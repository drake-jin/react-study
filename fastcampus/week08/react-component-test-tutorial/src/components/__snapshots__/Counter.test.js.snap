// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Counter.js decrease correctly 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
     + 
  </button>
  <button
    onClick={[Function]}
  >
     - 
  </button>
</div>
`;

exports[`src/components/Counter.js increase correctly 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    2
  </h2>
  <button
    onClick={[Function]}
  >
     + 
  </button>
  <button
    onClick={[Function]}
  >
     - 
  </button>
</div>
`;

exports[`src/components/Counter.js matches snapshot 1`] = `
<div>
  <h1>
    카운터
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
     + 
  </button>
  <button
    onClick={[Function]}
  >
     - 
  </button>
</div>
`;
